Plugin/DatatransPlugin.php

```diff
@@ -127,19 +135,15 @@ public function deposit(FinancialTransactionInterface $transaction, $retry)
     {
         try {
             $settlementRequest = $this->createSettlementRequest($transaction);
-
             $settlementResponse = $this->client->payComplete($settlementRequest);
             $this->throwUnlessSuccessPayComplete($settlementResponse);
-
-        } catch(\Exception $e) {
-            $this->throwFinancialTransaction($transaction, $e->getMessage());
+        } catch (Exception $e) {
+            $this->throwFinancialTransaction($transaction, $e->getMessage(), $settlementResponse->getResponseCode());
         }
 
-        $transaction->setReferenceNumber($settlementResponse->getRefno());
```

Comment:
Was this line deleted on purpose? No `Refno` for settlements?

Reply:
Yes! It has already been set in the previous transaction and will not be provided in the settlementResponse again. Would be nulled here.